src/metadata/ParamMetadata.ts

```diff
@@ -114,7 +114,14 @@ export class ParamMetadata {
         this.transform = args.transform;
         this.classTransform = args.classTransform;
         this.validate = args.validate;
-        this.targetType = args.explicitType ? args.explicitType : (Reflect as any).getMetadata("design:paramtypes", args.object, args.method)[args.index];
+        if (args.explicitType) {
+			this.targetType = args.explicitType
+		} else {
+			const metadata = (Reflect as any).getMetadata("design:paramtypes", args.object, args.method);
+			if (typeof metadata != "undefined") {
```

Comment:
Dont we have to set `this.targetType` to something if `metadata` is undefined? I am just wondering, cannot cause an other error later if we leave it `undefined`?

Reply:
It shouldn't, we have a check `if (this.targetType)` a few lines below.

Reply:
Ok. Then it's ok for me. However we should fix the indentation issue after merging.